File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let answer = 0
    let arr = ["aya", "ye", "woo", "ma"]
    
    for (let i = 0; i < babbling.length; i++) {
        let bab = babbling[i]
        
        for (let j=0; j < arr.length; j++) {
            if (bab.includes(arr[j].repeat(2))) {
                break
            }
            bab = bab.split(arr[j]).join(" ")
        }
        if (bab.split(" ").join("").length === 0) {
            answer ++
        }
    }
    return answer
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    valid_sounds = ["aya", "ye", "woo", "ma"]
    
    for bab_word in babbling:
        has_consecutive = False
        for sound in valid_sounds:
            if (sound * 2) in bab_word:
                has_consecutive = True
                break
        
        if has_consecutive:
            continue
            
        current_bab = bab_word
        for sound in valid_sounds:
            current_bab = current_bab.replace(sound, " ")
        
        if current_bab.replace(" ", "") == "":
            answer += 1
            
    return answer